pandas/io/parsers.py

```diff
@@ -1615,11 +1605,7 @@ def _get_name(icol):
                 return icol
 
             if col_names is None:
-                raise ValueError(
-                    ("Must supply column order to use {icol!s} as index").format(
-                        icol=icol
-                    )
-                )
+                raise ValueError(f"Must supply column order to use {icol:s} as index")
```

Comment:
`{icol:s}` is not the same as the original `{icol!s}`. If `icol` is not a string, this will fail, since `s` won't be a valid format, while `!s` calls `str()` and should always work.

Reply:
was unaware of this difference.  Thanks